### PR TITLE
added google analytics

### LINKED ISSUE #58: Add google analytics to uptane.github.io
It looks like this is possible. 
https://stackoverflow.com/questions/17207458/how-to-add-google-analytics-tracking-id-to-github-pages

### PR DESCRIPTION
**Pull request by devilkiller-ag**
# Summary of Changes
Added Plugin Configuration for Google Analytics.

## Related Issue
Closes #58

## Checklist
- [x] I have read and followed the [project's contribution guidelines](https://github.com/secure-systems-lab/lab-guidelines/blob/master/dev-workflow.md), including code style and commit message conventions.
- [x] My code is well-documented, and I've updated relevant documentation.
- [x] I have added or updated test cases to ensure the code's functionality.
- [x] I have tested these changes on my local environment.
- [x] All tests pass, and there are no new linting errors.
- [x] I have reviewed and proofread my code and the changes.
- [x] The branch is up-to-date with the base branch.

## Reviewer(s)
@hexsecs

### CODE REVIEW

File: docusaurus.config.js [hunk 1]
```diff
@@ -80,6 +80,15 @@ const config = {
...
 
+  plugins: [
+    [
+      '@docusaurus/plugin-google-gtag',
+      {
+        trackingID: 'G-G856J3S8ZG',
+        anonymizeID: true,
+      },
+    ],
+  ],
```
Comment (hexsecs) [inline docusaurus.config.js:91]: We are using the @docusaurus/preset-classic so I think you just follow the instructions for the preset options
https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-google-gtag

File: docusaurus.config.js [hunk 2]
```diff
@@ -80,6 +80,15 @@ const config = {
     ],
   ],
 
+  plugins: [
+    [
+      '@docusaurus/plugin-google-gtag',
+      {
+        trackingID: 'G-G856J3S8ZG',
+        anonymizeID: true,
```
Comment (hexsecs) [inline docusaurus.config.js:88]: This should be anonymizeIP not anonymizeID

File: package.json [hunk 3]
```diff
@@ -16,6 +16,7 @@
   "dependencies": {
     "@docusaurus/core": "^2.4.3",
     "@docusaurus/plugin-content-blog": "^2.4.3",
+    "@docusaurus/plugin-google-gtag": "^2.4.3",
```
Comment (hexsecs) [inline package.json:19]: This line isn't needed if using the classic preset.

@Abhijay007 - see previous comment. Do we have an implementation preference?

**Review by hexsecs — CHANGES_REQUESTED** [review]
Please see feedback inline.

File: docusaurus.config.js (hunk 1 above)
Comment (Abhijay007) [reply docusaurus.config.js:91]: Ya I do think as if we are using `@docusaurus/preset-classic ` and they mentioned that there is no need to install this plugin as a dependency, then we can skip this step and use it as a preset option as mentioned in the documentation

Comment (devilkiller-ag) [reply docusaurus.config.js:91]: Hi, yeah I read about the implementation for `@docusaurus/preset-classic `. However, I was a bit confused between `@docusaurus/preset-classic` and just `classic` on line 46 if they were the same or not. So I used plugin for safety. I will now implement as per `@docusaurus/preset-classic `. Should I also change `classic` to `@docusaurus/preset-classic ` on line 46?

File: docusaurus.config.js (hunk 2 above)
Comment (devilkiller-ag) [reply docusaurus.config.js:88]: Yeah Thanks for mentioning. I think it was changed due to intellisense. I will resolve this.

File: package.json (hunk 3 above)
Comment (devilkiller-ag) [reply package.json:19]: Yeah if we are using `@docusaurus/preset-classic ` preset then we don't need to install `@docusaurus/plugin-google-gtag`

**Comment by devilkiller-ag** [comment]
Okay I will keep things as per the `@docusaurus/preset-classic`. Just confirm if I should change `classic` to `@docusaurus/preset-classic` on line 46 in the docusaurus.config.js file?

File: docusaurus.config.js [hunk 4]
```diff
@@ -76,11 +76,14 @@ const config = {
         theme: {
           customCss: require.resolve('./src/css/custom.css'),
         },
+        gtag: {
+          trackingID: 'G-G856J3S8ZG',
+          anonymizeIP: true,
+        }
```
Comment (hexsecs) [inline docusaurus.config.js:82]: Make sure you have the trailing comma as is the rest of the style for the site. It helps with maintainability.

Comment (hexsecs) [inline docusaurus.config.js]:
```suggestion
        },
```

File: docusaurus.config.js [hunk 5]
```diff
@@ -76,11 +76,14 @@ const config = {
...
       }),
-    ],
+    ]
```
Comment (hexsecs) [inline docusaurus.config.js]:
```suggestion
    ],
```

File: package.json [hunk 6]
```diff
@@ -43,4 +43,4 @@
   "engines": {
     "node": ">=16.14"
   }
-}
\ No newline at end of file
+}
```
Comment (hexsecs) [inline package.json:46]: No change necessary here. Please revert your changes back.

**Review by hexsecs — CHANGES_REQUESTED** [review]
Looks good. Two things before we can merge...

1) Please fix the couple of issues with the trailing commas and whitespace (inline in the review)
2) Your branch is out of date. you will want to pull in the latest changes from uptane into your branch.
image

**Review by hexsecs — APPROVED** [review]

**Comment by hexsecs** [comment]
It works! 
image

**Comment by devilkiller-ag** [comment]
Awesome 🎉

**Pull request merged**